ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java

```diff
@@ -659,30 +660,76 @@ public void setMaxLength(ReactEditText view, @Nullable Integer maxLength) {
+    } else if ("tel-national".equals(autoCompleteType)) {
+      setAutofillHints(view, HintConstants.AUTOFILL_HINT_PHONE_NATIONAL);
+    } else if ("tel-device".equals(autoCompleteType)) {
+      setAutofillHints(view, HintConstants.AUTOFILL_HINT_PHONE_NUMBER_DEVICE);
+    } else if ("postal-address".equals(autoCompleteType)) {
+      setAutofillHints(view, HintConstants.AUTOFILL_HINT_POSTAL_ADDRESS);
+    } else if ("postal-address-country".equals(autoCompleteType)) {
+      setAutofillHints(view, HintConstants.AUTOFILL_HINT_POSTAL_ADDRESS_COUNTRY);
+    } else if ("postal-address-extended".equals(autoCompleteType)) {
+      setAutofillHints(view, HintConstants.AUTOFILL_HINT_POSTAL_ADDRESS_EXTENDED_ADDRESS);
+    } else if ("postal-address-extended-postal-code".equals(autoCompleteType)) {
+      setAutofillHints(view, HintConstants.AUTOFILL_HINT_POSTAL_ADDRESS_EXTENDED_POSTAL_CODE);
+    } else if ("postal-address-locality".equals(autoCompleteType)) {
+      setAutofillHints(view, HintConstants.AUTOFILL_HINT_POSTAL_ADDRESS_LOCALITY);
+    } else if ("postal-address-region".equals(autoCompleteType)) {
+      setAutofillHints(view, HintConstants.AUTOFILL_HINT_POSTAL_ADDRESS_REGION);
     } else if ("street-address".equals(autoCompleteType)) {
-      setAutofillHints(view, View.AUTOFILL_HINT_POSTAL_ADDRESS);
+      setAutofillHints(view, HintConstants.AUTOFILL_HINT_POSTAL_ADDRESS_STREET_ADDRESS);
     } else if ("postal-code".equals(autoCompleteType)) {
-      setAutofillHints(view, View.AUTOFILL_HINT_POSTAL_CODE);
+      setAutofillHints(view, HintConstants.AUTOFILL_HINT_POSTAL_CODE);
     } else if ("cc-number".equals(autoCompleteType)) {
-      setAutofillHints(view, View.AUTOFILL_HINT_CREDIT_CARD_NUMBER);
+      setAutofillHints(view, HintConstants.AUTOFILL_HINT_CREDIT_CARD_NUMBER);
+    } else if ("sms-otp".equals(autoCompleteType)) {
+      setAutofillHints(view, HintConstants.AUTOFILL_HINT_SMS_OTP);
     } else if ("cc-csc".equals(autoCompleteType)) {
-      setAutofillHints(view, View.AUTOFILL_HINT_CREDIT_CARD_SECURITY_CODE);
+      setAutofillHints(view, HintConstants.AUTOFILL_HINT_CREDIT_CARD_SECURITY_CODE);
     } else if ("cc-exp".equals(autoCompleteType)) {
-      setAutofillHints(view, View.AUTOFILL_HINT_CREDIT_CARD_EXPIRATION_DATE);
+      setAutofillHints(view, HintConstants.AUTOFILL_HINT_CREDIT_CARD_EXPIRATION_DATE);
+    } else if ("cc-exp-day".equals(autoCompleteType)) {
+      setAutofillHints(view, HintConstants.AUTOFILL_HINT_CREDIT_CARD_EXPIRATION_DAY);
     } else if ("cc-exp-month".equals(autoCompleteType)) {
-      setAutofillHints(view, View.AUTOFILL_HINT_CREDIT_CARD_EXPIRATION_MONTH);
+      setAutofillHints(view, HintConstants.AUTOFILL_HINT_CREDIT_CARD_EXPIRATION_MONTH);
     } else if ("cc-exp-year".equals(autoCompleteType)) {
-      setAutofillHints(view, View.AUTOFILL_HINT_CREDIT_CARD_EXPIRATION_YEAR);
+      setAutofillHints(view, HintConstants.AUTOFILL_HINT_CREDIT_CARD_EXPIRATION_YEAR);
```

Comment:
Since there are so many, maybe a `Map<String, String>` that maps the RN prop value to the `HintConstants` value?

Reply:
Agreed, I'll convert it to Map after I get the BUCK issue resolved.

Reply:
Done with this commit b9adfac6c7d8f59e855cc21d8e99375c99453c06